File: android/src/main/kotlin/tech/shmy/headless_webview/HeadlessWebviewManager.kt
import android.app.Activity
import android.content.Context
import io.flutter.plugin.common.MethodChannel
import tech.shmy.headless_webview.HeadlessWebview

class HeadlessWebviewManager {
    companion object {
        var webViews: HashMap<Int, HeadlessWebview> = HashMap()
        fun run(id: Int, url: String, channel: MethodChannel, context: Context, activity: Activity?): Int {
            val headlessWebview = HeadlessWebview(id, url, channel, context, activity)
            webViews[id] = headlessWebview
            return id
        }
    }
}

File: ios/Classes/HeadlessWebviewManager.swift
//
//  HeadlessWebviewManager.swift
//  headless_webview
//
//  Created by 王超 on 2022/1/29.
//

import Foundation
class HeadlessWebviewManager: NSObject {
    static var webViews: [Int: HeadlessWebview] = [:]
    
    public static func run(id: Int, url: String, channel: FlutterMethodChannel) -> Int {
       
        let headlessWebview = HeadlessWebview(channel: channel, id: id)
        webViews[id] = headlessWebview
        headlessWebview.load(url: url)
        return id
      
    }
}
